### PR TITLE
BIP141 script failure clarifications

### PR DESCRIPTION
**Pull request by jl2012**
Footnotes to clarify some special conditions which witness scripts will fail @sipa @instagibbs @gmaxwell

### CODE REVIEW

**Comment by instagibbs** [comment]
great, utACK

File: bip-0141.mediawiki [hunk 1]
```diff
@@ -98,9 +98,9 @@ If the version byte is 0, and the witness program is 32 bytes:
...
 
-If the version byte is 0, but the witness program is neither 20 nor 32 bytes, the script must fail.
+If the version byte is 0, but the witness program is neither 20 nor 32 bytes, the script must fail.<ref>For example, a scriptPubKey with OP_0 followed by a 40-byte non-zero data push will fail due to incorrect program size. However, a scriptPubKey with OP_0 followed by a 41-byte non-zero data push will pass, since it is not considered to be a witness program</ref>
 
-If the version byte is 1 to 16, no further interpretation of the witness program or witness happens, and there is no size restriction for the witness. These versions are reserved for future extensions.
+If the version byte is 1 to 16, no further interpretation of the witness program or witness happens, and there is no size restriction for the witness. These versions are reserved for future extensions.<ref>For backward compatibility, for any version byte from 0 to 16, the script must fail if the witness program has a <code>CastToBool</code> value of zero. However, having a hash like this is a successful preimage attack against the hash function, and the risk is negligible.</ref>
```
Comment (sipa) [inline bip-0141.mediawiki]: Maybe say: no *further* size restriction for the witness, besides the 40-byte limit.

Comment (instagibbs) [inline bip-0141.mediawiki]: This is referring to the witnessScript, AFAIK, not the witness program.

Comment (sipa) [inline bip-0141.mediawiki]: Oh, indeed.

Comment (instagibbs) [inline bip-0141.mediawiki]: I had the exact same misinterpretation you did a couple days ago. @jl2012 could we just spell it out?

Comment (jl2012) [inline bip-0141.mediawiki]: @sipa @instagibbs revised

**Comment by jl2012** [comment]
@sipa is it ok now?

**Comment by luke-jr** [comment]
@jl2012 You are an Author of this BIP, so any PRs you open will automatically get merged... so please don't open PRs until you are certain of a change.

In this case, I will wait for an explicit ACK since it sounds like you're checking with @sipa before desiring it to be merged.

**Comment by sipa** [comment]
ACK

**Pull request merged**